libraries/bezier-rs/src/subpath/solvers.rs

```diff
@@ -59,6 +59,62 @@ impl Subpath {
 
 		intersection_t_values
 	}
+
+	/// Return the min and max corners that represent the bounding box of the subpath.
+	pub fn bounding_box(&self) -> Option<[DVec2; 2]> {
+		let bounding_boxes = self.iter().map(|bezier| bezier.bounding_box()).collect::<Vec<[DVec2; 2]>>();
+		if bounding_boxes.is_empty() {
+			// There is no bounding box for empty subpath.
+			None
+		} else {
+			let mut min_point = bounding_boxes[0][0];
+			let mut max_point = bounding_boxes[0][1];
+			for bounding_box in bounding_boxes {
+				// We take min of mins and max of maxes.
+				min_point = min_point.min(bounding_box[0]);
+				max_point = max_point.max(bounding_box[1]);
+			}
+			Some([min_point, max_point])
+		}
+	}
+
+	/// Returns two lists of `t`-values representing the local extrema of the `x` and `y` parametric subpaths respectively.
+	/// The list of `t`-values returned are filtered such that they fall within the range `[0, 1]`.
+	pub fn local_extrema(&self) -> [Vec<f64>; 2] {
+		let (mut x_extremas, mut y_extremas) = (Vec::new(), Vec::new());
```

Comment:
Consider also use `.fold()` here too, but since this doesn't have the emptiness check if/else like above, it's less important to change this one.